crates/driver/src/domain/liquidity/zeroex.rs

```diff
@@ -1,5 +1,68 @@
-/// A signed 0x Protocol Limit Order [^1].
-///
-/// [^1]: <https://docs.0x.org/limit-orders-advanced-traders/docs/introduction>
+use {
+    crate::domain::{eth, liquidity},
+    anyhow::anyhow,
+    contracts::IZeroEx,
+    ethcontract::Bytes,
+    primitive_types::{H160, H256, U256},
+};
+
 #[derive(Clone, Debug)]
-pub struct LimitOrder {}
+pub struct LimitOrder {
+    pub order: Order,
+    pub zeroex: IZeroEx,
+}
+
+#[derive(Clone, Debug)]
+pub struct Order {
+    pub maker: H160,
+    pub taker: H160,
+    pub sender: H160,
+    pub maker_token: H160,
+    pub taker_token: H160,
+    pub maker_amount: u128,
+    pub taker_amount: u128,
+    pub taker_token_fee_amount: u128,
+    pub fee_recipient: H160,
+    pub pool: H256,
+    pub expiry: u64,
+    pub salt: U256,
+    pub signature_type: u8,
+    pub signature_r: H256,
+    pub signature_s: H256,
+    pub signature_v: u8,
+}
+
+impl LimitOrder {
+    pub fn to_interaction(&self, input: &liquidity::MaxInput) -> anyhow::Result<eth::Interaction> {
+        let method = self.zeroex.fill_or_kill_limit_order(
+            (
+                self.order.maker_token,
+                self.order.taker_token,
+                self.order.maker_amount,
+                self.order.taker_amount,
+                self.order.taker_token_fee_amount,
+                self.order.maker,
+                self.order.taker,
+                self.order.sender,
+                self.order.fee_recipient,
+                Bytes(self.order.pool.0),
+                self.order.expiry,
+                self.order.salt,
+            ),
+            (
+                self.order.signature_type,
+                self.order.signature_v,
+                Bytes(self.order.signature_r.0),
+                Bytes(self.order.signature_s.0),
+            ),
+            input.0.amount.0.as_u128(),
```

Comment:
This should probably do `.try_into().context("executed amount does not fit into u128")` because the `liquidity::MaxInput` type theoretically allows bigger amounts than `0x` can handle.